src/status_im/ui/screens/offline_messaging_settings/events.cljs

```diff
@@ -16,6 +18,45 @@
                               {:dispatch [:logout]}
                               (accounts-events/update-settings (assoc-in settings [:wnode network] wnode))))))
 
+(handlers/register-handler-fx
+ :save-wnode-transaction
+ (fn [{:keys [db] :as cofx} [_ wnode hash]]
+   (let [network     (ethereum/network->chain-keyword (:network db))
+         settings    (get-in db [:account/account :settings])
+         transaction (get-in db [:wallet :transactions hash])]
+     (when (not (nil? transaction))
+       (handlers-macro/merge-fx cofx
+                                {:dispatch [::save-wnode wnode hash]}
+                                (accounts-events/update-settings (assoc-in settings [:wnode-payment network wnode] hash)))))))
+
+(handlers/register-handler-fx
+ ::pay-wnode
+ (fn [{{:keys [web3] :as db} :db} [_ wnode]]
+   (let [network                         (ethereum/network->chain-keyword (:network db))
+         {:keys [address amount symbol]} (get-in db [:inbox/wnodes network wnode :payment])
+         contract                        (:address (tokens/symbol->token network symbol))
+         from                            (get-in db [:account/account :address])
+         gas                             (ethereum/estimate-gas symbol)
+         gas-price                       (money/->wei :gwei 5)]
+     {:db (update-in db [:wallet :send-transaction] assoc :waiting-signal? false)}
+     (erc20/transfer web3
+                     contract
+                     from
+                     address
+                     amount
+                     {:gas gas :gasPrice gas-price}
+                     #(dispatch [:save-wnode-transaction wnode %2])))))
+
+(handlers/register-handler-fx
+ :select-wnode
+ (fn [{:keys [db]} [_ wnode]]
+   (let [network    (ethereum/network->chain-keyword (:network db))
+         payment    (get-in db [:inbox/wnodes network wnode :payment])
+         payment-tx (get-in db [:account/account :settings :wnode-payment network wnode])]
+     (if (or (nil? payment) (not (nil? payment-tx)))
```

Comment:
Nil pruning

Reply:
What exactly is nil pruning? I have switched to using `contains?` if that clarifies it?